equipmanager.lic

```diff
@@ -395,4 +397,82 @@ class EquipmentManager
       stow_helper("stow my #{weapon_name}", weapon_name, 'You put', 'You should unload', 'You easily strap', 'You secure your')
     end
   end
+
+  # Given an item and a description (ajd + noun or just noun)
+  # then returns true if the item matches, else false.
+  def item_matches_desc?(item, desc)
+    return false unless item && desc
+    check_item_matches_regex_pr4484?(item, desc)
+    return (item.short_regex =~ desc) != nil
+  end
+
+  # ---------------------------------------------------------------
+
+  private
+
+  # Given an item object and a description text
+  # then returns true or false if the item's short_regex matches.
+  # This is a precursor to https://github.com/rpherbig/dr-scripts/pull/4484
+  # to begin checking if user's config for items need to be
+  # updated before that pull request is merged.
+  # If yes then we alert the user via this method.
+  # This method should not be used outside this class
+  # as it is intended to be removed once PR#4484 is merged.
+  def check_item_matches_regex_pr4484?(item, name_to_match)
+
+    # The old regex matches words that start with the adj/noun
+    # The issue is that can lead to false positives where "black book" matches "blackened bookshelf"
+    old_regex = item.short_regex
+    old_matches = (old_regex =~ name_to_match) != nil
+
+    # The new regex matches words exactly in the config
+    # by adding `\b` word boundary token around the words.
+    # Assuming `name_to_match` comes from DRC.left_hand or DRC.right_hand
+    # then the new regex requires the YAML config to match the adj/noun
+    # as it's known by the game, not shorthand words that infer a match.
+    new_regex = item.adjective ? /\b#{item.adjective}\b.*\b#{item.name}\b/i : /\b#{item.name}\b/i
+    new_matches = (new_regex =~ name_to_match) != nil
+
+    # Remove leading article and help text for closed containers.
+    name_to_match = name_to_match.gsub(/^(a|an|some)\s+/, '').gsub(/\s+\(closed\)/, '')
+
+    if @debug_mode_em
+      echo "check_item_matches_regex: #{@check_item_matches_regex}"
+      echo "item: #{name_to_match}"
+      echo "old regex: #{old_regex}"
+      echo "old matches? #{old_matches}"
+      echo "new regex: #{new_regex}"
+      echo "new matches? #{new_matches}"
+    end
+
+    # If the old regex matches but the new regex doesn't then
+    # notify user of the discrepancy.
+    if @check_item_matches_regex && old_matches && !new_matches
```

Comment:
Note to self, move `@check_item_matches_regex` to line 405 so only call this method if the setting is true